docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,406 @@
+
+```
+syntax = "proto3"
+
+message WakuMessageArchiveMetadata {
+  uint8 version = 1
+  uint64 from = 2
+  uint64 to = 3
+  repeated string contentTopic = 4
+}
+
+message WakuMessageArchive {
+  uint8 version = 1
+  WakuMessageArchiveMetadata metadata = 2
+  repeated WakuMessage messages = 3 // `WakuMessage` is provided by 14/WAKU2-MESSAGE
+  bytes padding = 4
+}
+```
+
+## Message history archive index
+
+Community owner nodes MUST provide message archives for the entire community history. Each individual archive only contains a subset of the complete history, that is, data for a time range of seven days, and all message history archives are concatenated into a single file as byte string (see [Ensuring reproducible data pieces](#ensuring-reproducible-data-pieces)).
+
+Community owner nodes MUST create a message history archive index (`WakuMessageArchiveIndex`) with metadata that allows receiving nodes to only fetch the message history archives they are interested in.
+
+### WakuMessageArchiveIndex
+
+A `WakuMessageArchiveIndex` is a map where the key is the KECCAK-256 hash of the `WakuMessageArchiveIndexMetadata` derived from a 7-day archive and the value is an instance of that `WakuMessageArchiveIndexMetadata` corresponding to that archive.
+
+The `offset` field MUST contain the position at which the message history archive starts in the byte string of the total message archive data. This MUST be the sum of the length of all previously created message archives in bytes (see [Creating message archive torrents](#creating-message-archive-torrents)).
+
+```
+syntax = "proto3"
+
+message WakuMessageArchiveIndexMetadata {
+  uint8 version = 1
+  WakuMessageArchiveMetadata metadata = 2
+  uint64 offset = 3
+  uint64 num_pieces = 4
+}
+
+message WakuMessageArchiveIndex {
+  map<string, WakuMessageArchiveIndexMetadata> archives = 1
+}
+```
+
+The community owner node MUST update the `WakuMessageArchiveIndex` every time it creates one or more `WakuMessageArchive`s and bundle it into a new torrent (**TODO: see section**).
+For every created `WakuMessageArchive`, there MUST be a `WakuMessageArchiveIndexMetadata` entry in the `archives` field `WakuMessageArchiveIndex`.
+
+## Creating message archive torrents
+
+Community owner nodes MUST create a torrent file ("torrent") containing metadata to all message history archives. To create a torrent file, and later serve the message archive data in the BitTorrent network, community owner nodes MUST store the necessary data in dedicated files on the file system.
+
+A torrent's source folder MUST contain the following two files:
+
+- `data` - Contains all protobuf encoded message history archives concatenated in ascending order
+- `index` - Contains the protobuf encoded message history archive index
+
+Community owner nodes SHOULD store these files in a dedicated folder that is identifiable via the community id.
```

Comment:
Community id is not defined

Reply:
You mean it should be added to the terminology section?

Reply:
Yes, indeed

Reply:
> dedicated folder that is identifiable via the community id.

Can you please elaborate on this? the folder should have the same name as the community id?